### PR TITLE
Socket Can extension

### PR DESCRIPTION
**Pull request by destogl**
Hi guys,

this is not ready to merge yet, but just not to (possibly) double efforts. I started to write connection from cob_generic_can for SocketCAN, this is using [socketcan_interface](https://github.com/ros-industrial/ros_canopen/tree/indigo-devel/socketcan_interface).

If there are any comments and ideas you are welcome.

### CODE REVIEW

File: cob_generic_can/CMakeLists.txt [hunk 1]
```diff
@@ -1,27 +1,30 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(cob_generic_can)
 
-find_package(catkin REQUIRED COMPONENTS cob_utilities libntcan libpcan)
+find_package(catkin REQUIRED COMPONENTS socketcan_interface cob_utilities libntcan libpcan)
```
Comment (fmessmer) [inline cob_generic_can/CMakeLists.txt]: please respect alphabetical order throughout this file (find_package, CATKIN_DEPENDS,...)

File: cob_generic_can/CMakeLists.txt [hunk 2]
```diff
@@ -1,27 +1,30 @@
...
-  CATKIN_DEPENDS cob_utilities libntcan libpcan
+  CATKIN_DEPENDS socketcan_interface cob_utilities libntcan libpcan
   INCLUDE_DIRS common/include
-  LIBRARIES ${PROJECT_NAME}_peaksysusb ${PROJECT_NAME}_peaksys ${PROJECT_NAME}_esd
+  LIBRARIES ${PROJECT_NAME}_peaksysusb ${PROJECT_NAME}_peaksys ${PROJECT_NAME}_esd ${PROJECT_NAME}_socketcan
 )
 
 ### BUILD ###
 include_directories(common/include ${catkin_INCLUDE_DIRS})
+include_directories(../ros_canopen/socketcan_interface/include ${catkin_INCLUDE_DIRS})
```
Comment (fmessmer) [inline cob_generic_can/CMakeLists.txt]: NoGo: relative path!
Is this needed, if you have `socketcan_interface` as a dependency and `${catkin_INCLUDE_DIRS}` added?
Also, `common_include` has been removed....why?

Comment (destogl) [inline cob_generic_can/CMakeLists.txt]: You are right, this doesn't make any sense. I will check it....

File: cob_generic_can/common/include/cob_generic_can/CanItf.h [hunk 3]
```diff
@@ -82,10 +83,11 @@ class CanItf
 public:
 	enum CanItfType {
 		CAN_PEAK = 0,
-		CAN_PEAK_USN = 1,
+		CAN_PEAK_USB = 1,
```
Comment (fmessmer) [inline cob_generic_can/common/include/cob_generic_can/CanItf.h:86]: I see that this has been a type before...but has it been used with `CAN_PEAK_USN` before somewhere? It needs to be changed there too then!

Comment (destogl) [inline cob_generic_can/common/include/cob_generic_can/CanItf.h:86]: For that should probably be checked cob_driver packaged and possibly others....

Comment (fmessmer) [inline cob_generic_can/common/include/cob_generic_can/CanItf.h:86]: from a grep in `cob_driver` it seems nowhere else being used...

**Comment by fmessmer** [comment]
as otherwise this PR only adds functionality and has no effect on existing interfaces, I'm going to merge this...

**Comment by destogl** [comment]
Ok. I have a new student working on it. Until now compiles but I didn't manage to get answer from sensor until now.

**Pull request merged**